### PR TITLE
fix: Fix tap event being fired incorrectly on parent then child then parent

### LINKED ISSUE #9643: [WASM] UIElement.Tapped called in wrong sequence
### Current behavior

Please see attached and search MainPage.xaml.cs for '!!!' - I left a few comments.

[Tapped.zip](https://github.com/unoplatform/uno/files/9439404/Tapped.zip)

### Expected behavior

WASM should work as WindowsAppSDK

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #9643

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/NestedElementsTap_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+	[Test]
+	[AutoRetry]
+	public void Button_Inside_Border_Tap()
+	{
+		Run("UITests.Shared.Windows_UI_Input.PointersTests.Button_Inside_Border_Click_Event");
+		var button = _app.Marked("MyButton");
+		Assert.AreEqual(string.Empty, (string)button.GetDependencyPropertyValue("Tag"));
+		button.FastTap();
+		Assert.AreEqual("Hit MyButton.OnTapped\r\n", (string)button.GetDependencyPropertyValue("Tag"));
```
Comment (Youssef1313) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/NestedElementsTap_Tests.cs:24]: This looks weird. I used `\n` in the sample. I don't think it's correct to normalize it to `\r\n`, but will have to debug that separately and validate the behavior on Windows.

**Comment by DierkDroth** [comment]
(@Youssef1313 not related to this PR but still: kudos for your amazing work on UNO, I'm impressed by the speed you're addressing issues)

File: src/Uno.UI/UI/Xaml/UIElement.Pointers.cs [hunk 2]
```diff
@@ -1030,6 +1028,8 @@ internal bool OnPointerUp(PointerRoutedEventArgs args, BubblingContext ctx = def
 				}
 			}
 
+			handledInManaged |= SetPressed(args, false, ctx);
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/UIElement.Pointers.cs:1031]: Well I think will break the event sequencing: The `PointerUp` has to be raise before the `Tap`

**Comment by DierkDroth** [comment]
@Youssef1313 is there any chance this would be closed off?

**Pull request closed without merging**